### PR TITLE
Add char limit warn

### PR DESCRIPTION
**Pull request by WeberJulian**
- [x] Add language dependent char limits
- [x] Fix error on failed audio loading during training

### CODE REVIEW

File: TTS/tts/layers/xtts/tokenizer.py [hunk 1]
```diff
@@ -476,6 +476,28 @@ def __init__(self, vocab_file=None, preprocess=None):
                 self.preprocess = preprocess
 
             self.tokenizer = Tokenizer.from_file(vocab_file)
+    
+    def check_input_length(self, txt, lang):
+        char_limits = {
```
Comment (akx) [inline TTS/tts/layers/xtts/tokenizer.py:481]: For performance, this should probably be a global constant so it isn't recomputed for each call of this function?

Comment (WeberJulian) [reply TTS/tts/layers/xtts/tokenizer.py:481]: Good point, thx

File: TTS/tts/layers/xtts/tokenizer.py [hunk 2]
```diff
@@ -538,8 +538,45 @@ def __init__(self, vocab_file=None):
...
+
+    def preprocess_text(self, txt, lang):
+        if lang in ["en", "es", "fr", "de", "pt", "it", "pl", "ar", "cs", "ru", "nl", "tr", "zh-cn"]:
+            txt = multilingual_cleaners(txt, lang)
+            if lang == "zh-cn":
+                txt = chinese_transliterate(txt)
+        elif lang == "ja":
+            if self.katsu is None:
+                import cutlet
+                self.katsu = cutlet.Cutlet()
```
Comment (akx) [inline TTS/tts/layers/xtts/tokenizer.py:572]: [`@cached_property`](https://docs.python.org/3.12/library/functools.html#functools.cached_property)?

File: TTS/tts/layers/xtts/tokenizer.py [hunk 3]
```diff
@@ -535,11 +536,50 @@ def korean_cleaners(text):
...
+            "zh-cn": 82,
+            "ar": 166,
+            "cs": 186,
+            "ru": 182,
+            "nl": 251,
+            "tr": 226,
+            "ja": 71,
+            "hu": 224,
+            "ko": 95,
+        }
```
Comment (akx) [inline TTS/tts/layers/xtts/tokenizer.py:558]: Maybe move these right under `class VoiceBpeTokenizer:`? That'd make each instance share the same dict. (Also, should these be sorted?)

**Review by Edresson — APPROVED** [review]

File: TTS/tts/layers/xtts/tokenizer.py (hunk 3 above)
Comment (WeberJulian) [reply TTS/tts/layers/xtts/tokenizer.py:558]: Should only be one instance at all times.

File: TTS/tts/layers/xtts/tokenizer.py [hunk 4]
```diff
@@ -535,11 +536,50 @@ def korean_cleaners(text):
...
+        import cutlet
+        return cutlet.Cutlet()
+    
+    def check_input_length(self, txt, lang):
+        limit = self.char_limits.get(lang, 250)
+        if len(txt) > limit:
+            print(f"[!] Warning: The text length exceeds the character limit of {limit} for language '{lang}', this might cause truncated audio.")
+
+    def preprocess_text(self, txt, lang):
+        if lang in ["en", "es", "fr", "de", "pt", "it", "pl", "ar", "cs", "ru", "nl", "tr", "zh-cn"]:
```
[the remark this answers is not in the document]
Comment (WeberJulian) [reply TTS/tts/layers/xtts/tokenizer.py:571]: In principle you're right but this list is never gonna be long. A dozen nano second won't justify running the CI again.

**Pull request merged**